specifyweb/specify/tree_extras.py

```diff
@@ -659,4 +659,21 @@ def renumber_tree(table):
     from .models import datamodel, Sptasksemaphore
     tree_model = datamodel.get_table(table)
     tasknames = [name.format(tree_model.name) for name in ("UpdateNodes{}", "BadNodes{}")]
-    Sptasksemaphore.objects.filter(taskname__in=tasknames).update(islocked=False)
\ No newline at end of file
+    Sptasksemaphore.objects.filter(taskname__in=tasknames).update(islocked=False)
+
+def is_instance_of_tree_def_item(obj):
+    from specifyweb.specify.models import (
+        Geographytreedefitem,
+        Geologictimeperiodtreedefitem,
+        Lithostrattreedefitem,
+        Storagetreedefitem,
+        Taxontreedefitem,
+    )
```

Comment:
not familiar with python, but is it ok performance-wise to import this inside of is_instance_of_tree_def_item?
is this function in the hot-path? would this cause performance issues?

or can we not import these in the global scope as they are not defined yet?

Reply:
This was done in order to prevent a circular import error.

Luckily it should be fine performance-wise, Python efficiently handles imports by reusing already imported modules instead of reloading them, so placing import statements inside functions should not significantly impact performance.

But, I think I might go ahead and just change it to `import specifyweb.specify.models as spmodels` instead.